instance/tests/integration/integration_instance.py

```diff
@@ -81,6 +83,44 @@ def assert_swift_container_provisioned(self, instance):
         stat_result = stat_container(instance.swift_container_name)
         self.assertEqual(stat_result.read_acl, '.r:*')
 
+    def assert_mysql_db_provisioned(self, instance):
+        """
+        Verify that the MySQL database for the instance has been provisioned and can be
+        connected to with the credentials the instance provides.
+        """
+        for database in instance.mysql_databases:
+            connection = mysql.connect(
+                host=settings.INSTANCE_MYSQL_URL_OBJ.hostname,
+                user=database['user'],
+                passwd=instance._get_mysql_pass(database['user']),
+                port=settings.INSTANCE_MYSQL_URL_OBJ.port or 3306,
+            )
+            database_name = connection.escape_string(database['name']).decode()
+            cur = connection.cursor()
+            result = cur.execute(
+                "SELECT SCHEMA_NAME FROM INFORMATION_SCHEMA.SCHEMATA WHERE SCHEMA_NAME = %s",
+                (database_name,),
+            )
+            self.assertEqual(result, 1)
+
+    def assert_mongo_db_provisioned(self, instance):
+        """
+        Verify that the databases ostensibly created by the provisioning process actually were.
+        """
+        for db_name in instance.mongo_database_names:
+            mongo = pymongo.MongoClient(
+                host=settings.INSTANCE_MONGO_URL_OBJ.hostname,
+                port=settings.INSTANCE_MONGO_URL_OBJ.port or 27017
+            )
+            # Insert a document into the database, verifying that you can.
```

Comment:
This comment doesn't seem to reflect what the code is doing... It looks to me like it's only calling `db.authenticate()` but not doing any insertion?